src/net.cpp

```diff
@@ -3549,26 +3547,11 @@ void CConnman::ThreadOpenConnections(const std::vector<std::string> connect, CDe
                 continue;
             }
 
-            // Port validation in Dash has additional rules. Some networks are prohibited
-            // from using a non-default port while others allow any arbitary port so long
-            // it isn't a bad port (and in the case of masternodes, it matches its listen
-            // port)
-            const bool is_prohibited_port = [this, &addr, &isMasternode](){
-                if (!Params().AllowMultiplePorts()) {
-                    const uint16_t default_port{Params().GetDefaultPort(addr.GetNetwork())};
-                    assert(!IsBadPort(default_port)); // Make sure we never set the default port to a bad port
-                    return addr.GetPort() != default_port;
-                }
-                const bool is_bad_port{IsBadPort(addr.GetPort())};
-                if (isMasternode) {
-                    return addr.GetPort() != GetListenPort() || is_bad_port;
-                } else {
-                    return is_bad_port;
-                }
-            }();
+            const uint16_t default_port{Params().GetDefaultPort(addr.GetNetwork())};
+            assert(!IsBadPort(default_port)); // Make sure we never set the default port to a bad port
```

Comment:
maybe we just drop this now? or figure out a way to static assert it

Reply:
how about 8ee01a0688?